client/src/pages/pot-instance/components/FoodTimerList/FoodTimerList.tsx

```diff
@@ -44,24 +52,51 @@ const FoodTimerList: FC<Props> = ({ hotpotStart, addCookedPot }) => {
     usePotContent([...tempObj]);
   };
 
-  // function that handles the timer countdown for each object in foodTimerObj. timer stops when the counter reaches 0. counter is read from cookTime
   const handleTime = () => {
-    useHotPotDuration(hotPotDuration + 1);
+    // function that updates the hotpotduration timer based on current time.
+
+    const tempHotPotCurrent = Math.floor(Date.now() / 1000);
+    const newElapsedTime = tempHotPotCurrent - hotPotDuration.hotPotStartTime;
+    const tempHotPotObj = {
+      hotPotStartTime: hotPotDuration.hotPotStartTime,
+      hotPotElapsedTime: newElapsedTime
+    };
+    useHotPotDuration(tempHotPotObj);
+
+    // function that handles the timer countdown for each object in foodTimerObj. timer stops when the counter reaches 0. counter is read from the
+    // resulting calculation from endtime and currenttime
     Object.entries(potContent).forEach(([key, value]) => {
-      if (value.cookTime > 0) {
+      if (value.timeLeft > 0) {
         const tempObj = potContent;
         const currentObj = potContent[parseInt(key, 10)];
-        currentObj.cookTime -= 1;
+        currentObj.currentTime = Math.floor(Date.now() / 1000);
+        currentObj.timeLeft = currentObj.endTime - currentObj.currentTime;
         tempObj.splice(parseInt(key, 10), 1, currentObj);
         usePotContent(tempObj);
       }
```

Comment:
This is probably a rare case, but you should maybe consider hard setting the `timeLeft` to 0 if its under 0s.

Reply:
Good call! Updated this